src/snowflake/connector/auth/by_plugin.py

```diff
@@ -89,18 +59,37 @@ class AuthType(Enum):
 class AuthByPlugin(ABC):
     """External Authenticator interface."""
 
-    def __init__(self) -> None:
-        self._retry_ctx = AuthRetryCtx()
+    def __init__(
+        self,
+        timeout: int | None = None,
+        max_retry_attempts: int | None = None,
+        socket_timeout_override: int | None = None,
```

Comment:
let's rename this to `socket_timeout` and add docstring to describe the override behavior